src/overlay/soil.py

```diff
@@ -25,7 +36,18 @@ def __init__(self, pos, group):
 
 
 class SoilLayer:
-    def __init__(self, all_sprites, tmx_map, frames, sounds):
+    all_sprites: pygame.sprite.Group
+    level_frames: dict
+
+    soil_sprites: pygame.sprite.Group
+    water_sprites: pygame.sprite.Group
+    plant_sprites: pygame.sprite.Group
+
+    map: dict[tuple[int, int], Tile]
```

Comment:
I'd advise against using this name, as it shadows a built-in iterator (see https://docs.python.org/3/library/functions.html#map).
You could rename it to `tile_map` to differentiate it from said iterator